File: PaymentServer/Order.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Xml;

namespace PaymentServer
{
	/*
	kvitto behöver(enligt coop och Legend spelbutik AB):
	orginisationsnummer
	Datum
	Kassör
	Kvitto

	summering

	moms
	.
	%
	belopp
	exklusiv/Netto
	Inklusiv/Brutto
	.

	Enligt kvitto.org
	Enligt Bokföringslagen (5 kapitlet, 7 §) ska verifikationen eller kvittot innehålla uppgifter om:

	när kvittot sammanställts,
	när affärshändelsen inträffat,
	vad den avser,
	vilket belopp,
	vilken motpart,
	verifikationsnummer eller annan identifikation som hänvisar till den bokförda affärshändelsen.

	Vid ett köp brukar följande information finnas med på kvittot:
	Säljarens namn och adress
	Organisationsnummer
	Datum och klockslag
	Artikelnamn/tjänstens beteckning och antal (ev. m. tillhörande beskrivning eller begränsning)
	Belopp / pris
	Moms
	*/
	class Order
	{
		public ulong reciptCounter { get; set; }
		XmlDocument document;

		public Order(string order)
		{
			document = new XmlDocument();
			document.LoadXml(order);
		}

		public float get_total_price()
		{
			return float.Parse(document.GetElementsByTagName("total_price")[0].InnerText);
		}

		public string get_phonenumber()
		{
			return document.GetElementsByTagName("phonenr")[0].InnerText;
		}

		public string get_tablenumber()
		{
			return document.GetElementsByTagName("tableid")[0].InnerText;
		}

		public string get_email()
		{
			return document.GetElementsByTagName("email")[0].InnerText;
		}

		public string toRecipt()
		{
			reciptCounter++;
			string html = "" +
				"<body>" +
					"<h1>Beställning från bord " + get_tablenumber() + "</h1>" +
					"---------------------------------------------" +
					"<p>Kvitto: " + reciptCounter + "</p>" +
					"<p>Org Nr: 556657-3621</p>" +
					"<p>Datum: " + DateTime.UtcNow.ToLocalTime() + "</p>" +
					"<p>Kassör: SjälvBetalning via Bord " + get_tablenumber() + "</p>" +
					"---------------------------------------------";

			var items = document.GetElementsByTagName("item");
			foreach (XmlNode it in items)
			{
				html += "<div style=\"overflow: auto; width: 600px; margin: 20px; font-weight: bold; font-size: 1.7em; border: solid gray; border-width: 0 0 3px 0;\">"
					+ it.Attributes.GetNamedItem("name").InnerText;

				foreach (XmlNode child in it.ChildNodes)
				{
                    switch (child.Name)
                    {
						case "price":
							html += "<div style=\"float: right; margin-left: 10px; font-weight: normal;\">" + child.InnerText;
                            Console.WriteLine(child.InnerText);
							break;

                        case "gluten_free":
                            html += "<div style=\"float: right; margin-left: 10px; font-weight: normal;\">Glutenfri</div>";
                            break;
                        case "egg_free":
                            html += "<div style=\"float: right; margin-left: 10px; font-weight: normal;\">Äggfri</div>";
                            break;
                        case "milk_free":
                            html += "<div style=\"float: right; margin-left: 10px; font-weight: normal;\">Mjölkfri</div>";
                            break;

							case 
                        default:
                            continue;
                    }
                }

				html += "</div>";
			}

			html += "</body>";

			return html;
		}

		public string toHTML()
		{
			string html = "" +
				"<body>" +
					"<h1>Beställning från bord " + document.GetElementsByTagName("tableid")[0].InnerText + "</h1>";

			var items = document.GetElementsByTagName("item");
			for (int i = 0; i < items.Count; i++)
			{

				html += "<div style=\"overflow: auto; width: 600px; margin: 20px; font-weight: bold; font-size: 1.7em; border: solid gray; border-width: 0 0 3px 0;\">" + items[i].Attributes.GetNamedItem("name").InnerText;
				string notes = "";

				foreach (XmlNode child in items[i].ChildNodes)
				{
                    switch (child.Name)
                    {
                        case "gluten_free":
                            html += "<div style=\"float: right; margin-left: 10px; font-weight: normal;\">Glutenfri</div>";
                            break;
                        case "egg_free":
                            html += "<div style=\"float: right; margin-left: 10px; font-weight: normal;\">Äggfri</div>";
                            break;
                        case "milk_free":
                            html += "<div style=\"float: right; margin-left: 10px; font-weight: normal;\">Mjölkfri</div>";
                            break;
                        case "notes":
                            notes += "<p style=\"margin-left: 40px; font-size: 1.3em;\">";
                            notes += child.InnerText;
                            notes += "</p>";
                            break;
                        default:
                            continue;
                    }
                }

				html += "</div>";

				html += notes;

			}

			html += "</body>";

			return html;
		}
	}
}

File: PaymentServerJava/src/paymentserver/Order.java
package paymentserver;

import org.joda.time.DateTime;
import org.joda.time.DateTimeZone;
import org.w3c.dom.Document;
import org.w3c.dom.Node;
import org.w3c.dom.NodeList;

public class Order
{
	private static long receiptCounter;
	public long getReceiptCounter()
	{
		return receiptCounter;
	}
	public void setReceiptCounter(long value)
	{
		receiptCounter = value;
	}
	private Document document;



	public Order(Document doc)
	{
	    this.document = doc;
	}

	public float get_total_price()
	{
		return Float.parseFloat(document.getElementsByTagName("total_price").item(0).getTextContent());
	}

	public String get_phone_number()
	{
		return document.getElementsByTagName("phonenr").item(0).getTextContent();
	}

	public String get_table_number()
	{
		return document.getElementsByTagName("tableid").item(0).getTextContent();
	}

	public String get_email()
	{
		return document.getElementsByTagName("email").item(0).getTextContent();
	}

	public String toReceipt()
	{
		DateTimeZone sweden = DateTimeZone.forID("Europe/Stockholm");
		DateTime now = DateTime.now(sweden);

		String dateTime = now.toString().split("T")[0] + " "
				+ (now.toString().split("T")[1]).split("\\.")[0];

		setReceiptCounter(getReceiptCounter() + 1);

		StringBuilder html = new StringBuilder();
		html.append("<body>" + "<h1>Bryggargatan </h1>" + "---------------------------------------------" + "<p>Kvitto: ").append(getReceiptCounter()).append("</p>").append("<p>Org Nr: 556657-3621</p>").append("<p>Datum: ").append(dateTime).append("</p>").append("<p>Kassör: SjälvBetalning via Bord ").append(get_table_number()).append("</p>").append("---------------------------------------------");


		NodeList items = document.getElementsByTagName("item");

		html.append("<div style=\"overflow: auto; width: 600px; margin: 20px; font-weight: bold; font-size: 1.7em; border: solid gray; border-width: 0 0 3px 0;\"><div style=\"float: right;\">SEK</div></div>");

		for (int i = 0; i < items.getLength(); i++)
		{
			html.append("<div style=\"overflow: auto; width: 600px; margin: 20px; font-weight: bold; font-size: 1.7em; border: solid gray; border-width: 0 0 3px 0;\">").append(items.item(i).getAttributes().getNamedItem("name").getTextContent());

			Node node = items.item(i);
			for (int j = 0; j < node.getChildNodes().getLength(); j++)
			{
				switch (node.getChildNodes().item(j).getNodeName())
				{
					case "price":
						html.append("<div style=\"float: right; margin-left: 10px;\">").append(node.getChildNodes().item(j).getTextContent()).append(",00</div>");
						break;

					case "gluten_free":
						html.append("<div style=\"float: right; margin-left: 10px;\">Glutenfri</div>");
						break;
					case "egg_free":
						html.append("<div style=\"float: right; margin-left: 10px;\">Äggfri</div>");
						break;
					case "milk_free":
						html.append("<div style=\"float: right; margin-left: 10px;\">Mjölkfri</div>");
						break;
					default:
				}
			}

			html.append("</div>"); //closes order div
		}



		html.append("<div style=\"overflow: auto; width: 600px; margin: 20px; font-weight: bold; font-size: 1.7em; border: solid gray; border-width: 0 0 3px 0;\">ATT BETALA");
		html.append("<div style=\"float: right; margin-left: 10px;\">").append(String.format("%.2f", (double) get_total_price())).append("</div>");
		html.append("</div>");

		float moms = 0.12f;
		html.append("<div>");
		html.append("<div style=\"display: inline-block; font-weight: normal;\">Moms</div>");
		html.append("<div style=\"display: inline-block; margin-left: 8%; font-weight: normal;\">Belopp</div>");
		html.append("<div style=\"display: inline-block; margin-left: 8%; font-weight: normal;\">Netto</div>");
		html.append("<div style=\"display: inline-block; margin-left: 8%; font-weight: normal;\">Brutto</div>");
		html.append("</div>");

		html.append("<div>");
		html.append("<div style=\"display: inline-block; font-weight: normal;\">").append((int) (moms * 100)).append("%</div>");
		html.append("<div style=\"display: inline-block; margin-left: 10%; font-weight: normal;\">").append(Math.round(get_total_price() * moms * 100.0) / 100.0).append("</div>");
		html.append("<div style=\"display: inline-block; margin-left: 10%; font-weight: normal;\">").append(Math.round(get_total_price() * (1 - moms) * 100.0) / 100.0).append("</div>");
		html.append("<div style=\"display: inline-block; margin-left: 10%; font-weight: normal;\">").append(get_total_price()).append("</div>");
		html.append("</div>");

		html.append("</body>");

		return html.toString();
	}
}